roxie/ccd/ccdmain.cpp

```diff
@@ -409,6 +406,136 @@ class CHpccProtocolPluginCtx : implements IHpccProtocolPluginContext, public CIn
     }
 };
 
+static SocketEndpointArray topologyServers;
+static std::vector<RoxieEndpointInfo> myRoles;
+static std::vector<unsigned> farmerPorts;
+static std::vector<unsigned> slaveChannels;
+
+static bool splitarg(const char *arg, std::string &name, std::string &value)
+{
+    const char *eq = strchr(arg, '=');
+    if (eq)
+    {
+        name.append(arg, eq-arg);
+        value.append(eq+1);
+        return true;
+    }
+    else
+        return false;
+}
+
+
+void readStaticTopology()
+{
+    // If dynamicServers not set, we read a list of all servers form the topology file, and deduce which ones are on which channel
+    // and the total number of channels
+    static std::vector<RoxieEndpointInfo> allRoles;
```

Comment:
I don't think it should be static any longer.

Reply:
It should not